### ISSUE TITLE
Header is imcompatible with c++

### ISSUE THREAD

**Issue by hq6** [dbittman/waitfree-mpsc-queue#7]
The header is not c++ friendly.

### Reproducing the problem

Put the following program into a file called `main.cc`
```
#include "mpscq.h"
int main(){}
```

Compile with `g++ -c main.cc` and get multiple errors that look like this.

```
usr/lib/gcc/x86_64-linux-gnu/4.9/include/stdatomic.h:68:9: error: ‘_Atomic’ does not name a type
 typedef _Atomic __UINT_FAST32_TYPE__ atomic_uint_fast32_t;
         ^
/usr/lib/gcc/x86_64-linux-gnu/4.9/include/stdatomic.h:69:9: error: ‘_Atomic’ does not name a type
 typedef _Atomic __INT_FAST64_TYPE__ atomic_int_fast64_t;

```

**Comment by dbittman** [comment]
This is actually something that bothers me in general in C++. Since this is a compiler-provided header, I kinda wished it worked fine on both languages.

**Comment by hq6** [comment]
I agree that this is a silly compiler issue, but I think as our thread on the #8 shows, there are many solutions to achieving compatibility with C++, and I think it would make this implementation more broadly useable.
